pkg/dependencies/factory.go

```diff
@@ -274,13 +274,16 @@ func (f *Factory) WithTroubleshoot() *Factory {
 	return f
 }
 
-func (f *Factory) WithNetworking() *Factory {
+func (f *Factory) WithNetworking(clusterConfig *v1alpha1.Cluster) *Factory {
 	f.buildSteps = append(f.buildSteps, func() error {
 		if f.dependencies.Networking != nil {
 			return nil
 		}
-
-		f.dependencies.Networking = networking.NewCilium()
+		if clusterConfig.Spec.ClusterNetwork.CNI == "kindnetd" {
```

Comment:
is there a way to use the [constant](https://github.com/aws/eks-anywhere/pull/702/files#diff-e930b9692e6f9562202b02e50f1e6c4690205ca60ae79dd9dbb96c849e716459R304)

Reply:
`v1alpha1.Kindnetd`